### PR TITLE
Add pending results to clinical info for ecr viewer

### LINKED ISSUE #1104: Add Pending Results to eCR Viewer
## Action Requested
Add the "Pending Results" field to the eCR Viewer.
>eCR Viewer
>> Clinical Info
>>> Treatment Details
>>>> Plan of Treatment
>>>>> Pending Results

## Acceptance Criteria
Pending Results are displayed for eCRs that have the field.

## Testing
Unit testing will be required.

## Background & Context
This is likely a list with multiple labs included.

## Additional Information
We'll need to test to see whether or not this field is already being converted to FHIR, or if we'll need to complete both FHIR conversion and adding to the eCR Viewer.

### PR DESCRIPTION
**Pull request by BobanL**
# PULL REQUEST

## Summary
What changes are being proposed?
- Adding pending results to clinical info
- Refactor `formatTablesToJSON` to support tables that are not in a list.
- Add return types to `formatTablesToJSON` and `processTable`
- Fix minor codesmells
- Add `caption` tag to supported tags in fhir converter
- Add tests

Screen Shot 2024-04-05 at 2 12 13 PM

## Related Issue
Fixes #1104 

## Additional Information
Anything else the review team should know?

## Checklist

- [ ] If this code affects the other scrum team, have they been notified? (In Slack, as reviewers, etc.)

[//]: # (PR title: Remember to name your PR descriptively!)

### CODE REVIEW

File: containers/ecr-viewer/src/app/format-service.tsx [hunk 1]
```diff
@@ -161,27 +167,36 @@ export const formatString = (input: string): string => {
...
   const jsonArray: any[] = [];
-  doc.querySelectorAll("li").forEach((li) => {
-    const tables: any[] = [];
-    const resultId = li.getAttribute("data-id");
-    const resultName = li.childNodes[0].textContent?.trim() || "";
-    li.querySelectorAll("table").forEach((table) => {
-      tables.push(processTable(table));
+  const liArray = doc.querySelectorAll("li");
+  if (liArray.length > 0) {
+    doc.querySelectorAll("li").forEach((li) => {
```
Comment (lina-roth) [inline containers/ecr-viewer/src/app/format-service.tsx:184]: use liArray

File: containers/ecr-viewer/src/app/utils.tsx [hunk 2]
```diff
@@ -610,6 +700,10 @@ export const evaluateClinicalData = (
         mappings,
       ),
     },
+    {
+      title: "Plan of Treatment",
+      value: planOfTreatmentElement, //add scheduled order table here
```
Comment (lina-roth) [inline containers/ecr-viewer/src/app/utils.tsx:705]: remove comment

File: containers/ecr-viewer/src/app/view-data/components/AccordionContainer.tsx [hunk 3]
```diff
@@ -68,18 +68,22 @@ const AccordianContainer: React.FC<AccordionContainerProps> = ({
...
-            clinicalData.reasonForVisitDetails.availableData
-          }
-          activeProblemsDetails={
-            clinicalData.activeProblemsDetails.availableData
-          }
-          vitalData={clinicalData.vitalData.availableData}
-          immunizationsDetails={clinicalData.immunizationsDetails.availableData}
-          treatmentData={clinicalData.treatmentData.availableData}
-        />
+        <>
```
Comment (lina-roth) [inline containers/ecr-viewer/src/app/view-data/components/AccordionContainer.tsx:71]: remove tags

File: containers/ecr-viewer/src/app/view-data/components/ClinicalInfo.tsx [hunk 4]
```diff
@@ -7,6 +7,15 @@ import {
 import { SectionConfig } from "./SideNav";
 import React from "react";
 
+export type TableEntry = {
```
Comment (lina-roth) [inline containers/ecr-viewer/src/app/view-data/components/ClinicalInfo.tsx:10]: probably can delete this

File: containers/ecr-viewer/src/app/view-data/components/ClinicalInfo.tsx [hunk 5]
```diff
@@ -132,6 +141,7 @@ export const ClinicalInfo = ({
     );
   };
 
+  // TODO: write test for Plan of treatment without value 6100896d-b520-497c-b2fe-1c111c679274
```
Comment (lina-roth) [inline containers/ecr-viewer/src/app/view-data/components/ClinicalInfo.tsx:144]: check if this is still the case

**Comment by sarahtress** [comment]
@lina-roth only one request from me on this one - is it possible to make the columns not all even width? is there a setting that lets columns with more text take up more room and columns with less text to be narrrower? not sure if that makes sense in writing... but in this example, Type and Priority could be narrower columns and Name could be wider so the words aren't as cut off awkwardly in the first row

**Comment by BobanL** [comment]
> @lina-roth only one request from me on this one - is it possible to make the columns not all even width? is there a setting that lets columns with more text take up more room and columns with less text to be narrrower? not sure if that makes sense in writing... but in this example, Type and Priority could be narrower columns and Name could be wider so the words aren't as cut off awkwardly in the first row

There is a setting called `fixed` that forces the columns to be the same length. It was on in the before and it's off in the after.

Before:
image
 
After:
image


Does the after image look ok?

**Comment by sarahtress** [comment]
> > @lina-roth only one request from me on this one - is it possible to make the columns not all even width? is there a setting that lets columns with more text take up more room and columns with less text to be narrrower? not sure if that makes sense in writing... but in this example, Type and Priority could be narrower columns and Name could be wider so the words aren't as cut off awkwardly in the first row
> 
> There is a setting called `fixed` that forces the columns to be the same length. It was on in the before and it's off in the after.
> 
> Before: image
> 
> After: image
> 
> Does the after image look ok?

@BobanL I think this looks good enough for now!

File: containers/ecr-viewer/src/app/utils.tsx [hunk 6]
```diff
@@ -500,6 +507,65 @@ export const returnProblemsTable = (
...
+    const header = [
+      "Name",
+      "Type",
+      "Priority",
+      "Associated Diagnoses",
+      "Date/Time",
+    ];
+
+    return (
+      <Table
```
Comment (angelathe) [inline containers/ecr-viewer/src/app/utils.tsx:531]: Curious why you're building the table here -- does the `evaluateTable` function not work for the pending results table?

Comment (BobanL) [reply containers/ecr-viewer/src/app/utils.tsx:531]: pending results comes from the bundle as a HTML table and `evaluateTable` handles fhirPath results.

**Review by angelathe — APPROVED** [review]

**Pull request merged**